### ISSUE TITLE
OneSignal causing ANRs on JobIntentServices

### ISSUE THREAD

**Issue by janbolat** [OneSignal/OneSignal-Android-SDK#569]
Our app has Services that extend from android.support.v4.app.JobIntentService.
Our users are facing ANRs on Android 8.0 and 8.1 (99.99%).

I found a [similar](https://github.com/OneSignal/OneSignal-Android-SDK/issues/331) issue but it says that it was fixed. I am not sure if it relates to that problem.

```
"main" prio=5 tid=1 Blocked
  | group="main" sCount=1 dsCount=0 flags=1 obj=0x7355e620 self=0xe0d03000
  | sysTid=21308 nice=0 cgrp=default sched=0/0 handle=0xed4784bc
  | state=S schedstat=( 0 0 0 ) utm=34 stm=11 core=3 HZ=100
  | stack=0xff017000-0xff019000 stackSize=8MB
  | held mutexes=
  at com.onesignal.UserStateSynchronizer.getToSyncUserState (UserStateSynchronizer.java)
- waiting to lock <0x0c5e6014> (a com.onesignal.UserStatePushSynchronizer) held by thread 25
  at com.onesignal.UserStateSynchronizer.getRegistrationId (UserStateSynchronizer.java:31)
  at com.onesignal.OneSignalStateSynchronizer.getRegistrationId (OneSignalStateSynchronizer.java:123)
  at com.onesignal.OneSignal.internalFireIdsAvailableCallback (OneSignal.java:1701)
- locked <0x08ed30bd> (a java.lang.Class<com.onesignal.OneSignal>)
  at com.onesignal.OneSignal.access$2900 (OneSignal.java:80)
  at com.onesignal.OneSignal$15$1.run (OneSignal.java:1670)
  at android.os.Handler.handleCallback (Handler.java:789)
  at android.os.Handler.dispatchMessage (Handler.java:98)
  at android.os.Looper.loop (Looper.java:164)
  at android.app.ActivityThread.main (ActivityThread.java:6944)
  at java.lang.reflect.Method.invoke (Native method)
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:327)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1374)

"OSH_NetworkHandlerThread" prio=5 tid=25 TimedWaiting
  | group="main" sCount=1 dsCount=0 flags=1 obj=0x12f01df8 self=0xdcdb5200
  | sysTid=21347 nice=0 cgrp=default sched=0/0 handle=0xc8e94970
  | state=S schedstat=( 0 0 0 ) utm=0 stm=0 core=0 HZ=100
  | stack=0xc8d92000-0xc8d94000 stackSize=1038KB
  | held mutexes=
  at java.lang.Object.wait (Native method)
- waiting on <0x0ae3f644> (a java.lang.Object)
  at java.lang.Object.wait (Object.java:422)
  at java.lang.Thread.join (Thread.java:1268)
- locked <0x0ae3f644> (a java.lang.Object)
  at com.onesignal.OneSignalRestClient.makeRequest (OneSignalRestClient.java:106)
  at com.onesignal.OneSignalRestClient.putSync (OneSignalRestClient.java:82)
  at com.onesignal.UserStateSynchronizer.doPutSync (UserStateSynchronizer.java:255)
  at com.onesignal.UserStateSynchronizer.internalSyncUserState (UserStateSynchronizer.java:189)
  at com.onesignal.UserStateSynchronizer.syncUserState (UserStateSynchronizer.java:160)
- locked <0x0c5e6014> (a com.onesignal.UserStatePushSynchronizer)
  at com.onesignal.UserStateSynchronizer$NetworkHandlerThread$1.run (UserStateSynchronizer.java:69)
  at android.os.Handler.handleCallback (Handler.java:789)
  at android.os.Handler.dispatchMessage (Handler.java:98)
  at android.os.Looper.loop (Looper.java:164)
  at android.os.HandlerThread.run (HandlerThread.java:65)
```

`InitService:`
![image](https://user-images.githubusercontent.com/6511056/41702971-ad74d000-7553-11e8-868c-f1b521bbdc18.png)

`FetchDataService:`
![image](https://user-images.githubusercontent.com/6511056/41703004-c3084a64-7553-11e8-9836-79b3b9769b84.png)

`AlarmInitReceiver:`
![image](https://user-images.githubusercontent.com/6511056/41703158-323c207c-7554-11e8-8cdb-37cffca7c80d.png)

and more...

Environment:
```
implementation "com.android.support:support-v4:27.1.1"
implementation 'com.onesignal:OneSignal:3.9.1'
implementation "com.google.firebase:firebase-messaging:17.0.0"
```

**Comment by jkasten2** [comment]
@janbolat Thanks for the details. This does look like a different issue than the one you linked too.

As you highlighted in the logs it looks like there is a case where OneSignal can make the main thread wait on `OSH_NetworkHandlerThread` which looks to be the root of the cause here.

**Comment by zupodaniel** [comment]
Production app facing same issue here.

**Comment by yaronlevi** [comment]
We probably experiencing the same issue OSH_NetworkHandlerThread on our react native app.
We were at RN@0.49 and after we upgraded to RN@0.55 it started.
Also, we have "react-native-onesignal": "3.0.5".

![screen shot 2018-07-09 at 14 45 03](https://user-images.githubusercontent.com/5398977/42448970-af384470-8387-11e8-828d-0986b054adf1.png)

**Comment by rodrigon-rpc** [comment]
Also experiencing the same issue here on Android versions 8.0 and 8.1 in production!

**Comment by yaronlevi** [comment]
Once we upgraded to the latest react-native-onesignal the issue is gone 👍

**Comment by jkasten2** [comment]
I believe this issue was fixed in PR #607. Closing this issue as resolved.
